oauth2_client/providers/oauth2/views.py

```diff
@@ -0,0 +1,40 @@
+"""
+OAuth2Provider's views
+"""
+from django.http import HttpResponse
+
+from oauth2_client.exceptions import ImmediateHttpResponse
+
+
+class OAuth2View(object):
+    """
+    Abstract view to be extended when implementing views for your providers. It allows to break the
+    authentication flow immediately, by throwing `ImmediateHttpResponse` in extending code.
+    """
+    @classmethod
+    def build_view(cls):
+        def view(request, *args, **kwargs):
+            self = cls()
+            self.request = request
+            try:
+                return self.dispatch(request, *args, **kwargs)
+            except ImmediateHttpResponse as e:
+                return e.response
+
+        return view
+
+
+class OauthTestCallbackView(OAuth2View):
```

Comment:
please move to tests.
See https://docs.djangoproject.com/en/3.0/topics/testing/tools/#urlconf-configuration and https://github.com/Livit/Livit.Learn.LTI/blob/a2044da57beb98e89ee96f6d9671b53fbde3ac51/django_project/protocols/lti/tests/base.py#L44